sanic/blueprints.py

```diff
@@ -209,3 +237,27 @@ def register(self, app, options):
         self.middlewares = middleware
         self.exceptions = exception_handlers
         self.listeners = dict(listeners)
+
+    async def dispatch(self, *args, **kwargs):
+        where = kwargs.pop("where", {})
+        where.update({"blueprint": self.name})
+        kwargs["where"] = where
+        await asyncio.gather(
+            *[app.dispatch(*args, **kwargs) for app in self.apps]
+        )
+
+    def event(self, event: str):
+        events = set()
+        for app in self.apps:
+            signal = app.signal_router.name_index.get(event)
+            if not signal:
+                raise NotFound("Could not find signal %s" % event)
+            events.add(signal.ctx.event)
```

Comment:
When this BP is registered on multiple apps, does every app need to have a signal in the router? In the current code this will throw an error if _any_ of the apps doesn't have the signal. I think this an editing error, I think the correct logic is:
```suggestion
        events = set()
        for app in self.apps:
            signal = app.signal_router.name_index.get(event)
            if signal:
                events.add(signal.ctx.event)
```

Reply:
When you register a BP, all of its signals should be on the app. In the case where there are multiple apps, it should be on all of them. If it is missing from one, well, likely something went wrong somewhere.

I am fine making this change, but it still feels to me like something bad just happened that should be handled here.

Reply:
Ok, if it was your intention to do it that way, thats fine. It just looked like a copy&paste error when I was looking through the code, like that exception wasn't supposed to be there. (Eg, if the exception is thrown in the `for` loop, there is no way for the `NotFound` exception at the end of the function to be hit).

Reply:
I went back and forth on this myself a couple times. I think maybe you are right though that the final exception doesn't quite fit.

That exception would only be hit if there are no registered apps, in which caste the exception `NotFound` doesn't make sense.

Reply:
And, because we are accessing `self.apps` as a property, you would get an exception earlier than that before it ever reaches there.

I suppose I was thinking that we needed to catch when `events` was empty, but I am not seeing how it could be.

Anyone see something else?